ui/packages/app/src/AppRoutes.js

```diff
@@ -1,26 +1,39 @@
-import React from "react";
+import React, { useContext } from "react";
 import { Home, Project } from "./pages";
 import { ProjectCreation } from "./project_setting/ProjectCreation";
 import ProjectSetting from "./project_setting/ProjectSetting";
 import { Navigate, Route, Routes } from "react-router-dom";
+import { StreamlitPlaceholderPage } from "./streamlit_placeholder_page/StreamlitPlaceholderPage";
+import { ApplicationsContext } from "@caraml-dev/ui-lib";
 
-export const AppRoutes = () => (
-  <Routes>
-    {/* LANDING */}
-    <Route index element={<Home />} />
+export const AppRoutes = () => {
+  const { apps, isLoaded } = useContext(ApplicationsContext);
 
-    <Route path="projects">
-      {/* PROJECT LANDING PAGE */}
-      <Route path=":projectId" element={<Project />} />
-      {/* PROJECT SETTING */}
-      <Route path=":projectId/settings/*" element={<ProjectSetting />} />
-      {/* New Project */}
-      <Route path="create" element={<ProjectCreation />} />
-    </Route>
+  // If the apps have not been loaded yet, we do not render any of the app related routes - the additional streamlit
+  // apps need to be retrieved from the MLP API v2/applications endpoint first before we generate each route for them.
+  // Once those apps are loaded, AppRoutes will be re-rendered,
+  return isLoaded && (
+    <Routes>
+      {/* LANDING */}
+      <Route index element={<Home />} />
 
-    {/* DEFAULT */}
-    <Route path="*" element={<Navigate to="/pages/404" replace={true} />} />
-  </Routes>
-);
+      {apps?.map(app => !!app.streamlit_placeholder_page_config &&
+        <Route key={app.name} path={app.homepage} element={<StreamlitPlaceholderPage app={app} />} />)
+      }
```

Comment:
This is the only new addition to the `Routes` block, where we conditionally create a `Route` for each configured streamlit placeholder app.

Reply:
you can have multiple app?

Reply:
Yeah you can have multiple apps that use the iframe to display another UI.